### ISSUE TITLE
SVG Path element events not firing correctly

### ISSUE THREAD

**Issue by krityan** [ractivejs/ractive#3157]
## Description:

We have on-mouseenter/on-mouseleave events on path elements, in chrome/firefox these events fire correctly and are picked up by the respective listeners. In IE, however, the listener is never triggered (as shown by putting breakpoints of them).

The actual structure is an svg element, with a g element that contains path elements. The path elements are within a partial called line.

We have tried modifying the pointer-events attribute of the path elements to all, none and visible and none of them resolve this.

It seems to be a similar issue to what was fixed in #3122 but isn't resolved by the same fix in 0.9.8, it also worked as intended in 0.7.3

This is a simplified version of the component, the hoverOn and hoverOff listeners in the corresponding JS are not triggered in IE but are in chrome.

```
<div class='line-graph'>

    <svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"
        width=200
        height=200
        on-mousemove="mousemove"
        on-mouseleave="mouseleave"
        on-click="click">

        <g transform="translate(10, 10)">
            <defs>
                <clipPath id="js-clip">
                    <rect y="-10" width="180" height="180" />
                </clipPath>
            </defs>

            <g clip-path="url(#js-clip)"></g>

            {{#lines}}

                {{#if test}}
                <a xlink:href="www.google.com">
                    {{>line}}
                </a>
                {{else}}
                    {{>line}}
                {{/if}}

            {{/lines}}

            </g>
    </svg>
</div>

{{#partial line}}

    <path class="line"
        d="{{line.d}}"
        fill="none"
        stroke="{{line.colour}}"
        stroke-width="{{line.width}}"
        stroke-opacity="{{line.opacity}}"
        stroke-linejoin="round" />

    <g>
        <path class="hover-line"
        on-mouseenter="hoverOn"
        on-mouseleave="hoverOff"
        d="{{line.d}}"
        fill="none"
        stroke="#ffffff"
        stroke-width="8"
        stroke-opacity="0" />
    </g>

{{/partial}}
```

## Versions affected:

Worked fine in 0.7.3, noticed first in 0.9.3, still present in 0.9.10.

## Platforms affected:

IE 11

**Comment by evs-chris** [comment]
Does it work if you turn off [event delegation](https://ractive.js.org/api/#delegate) from your main instance e.g. `new Ractive({ delegate: false, ... })`?

**Comment by krityan** [comment]
No luck here sadly, all our components extend from a base component so I added `delegate: false` to that and also to the place where the first base component is initialised as: `Ractive.components.BaseComponent({....});` with neither resolving the issue. This was done on 0.9.10.

**Comment by evs-chris** [comment]
Poking at it a bit more, it seems that `SVGSVGElement` in IE doesn't have `onmouseenter` and `onmouseleave` properties, which is... weird. If you force the properties to exist, then listening carries on as normal and all works. I don't _see_ any special casing code in 0.7 for that, but I'll see if I can figure out why it works there to move it into 0.9.

**Comment by evs-chris** [comment]
Well, that didn't take too long to find. 0.7 also checks the `window` for supported DOM events, so IEs specialness is covered. This should now be fixed in edge. Thanks for the report!

**Comment by krityan** [comment]
Confirming this is resolved in 1.0.0 build 110.
